### PR TITLE
Add Rotational Cipher module

### LINKED ISSUE #150: Add Rotational cipher module

### PR DESCRIPTION
**Pull request by rizemon**
Fixes #150

### CODE REVIEW

File: src/main/java/decodex/modules/rot/RotEncoder.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+    public RotEncoder(int rotateOffset) {
+        super(MODULE_NAME, MODULE_DESCRIPTION);
+        this.rotateOffset = rotateOffset;
+    }
+
+    /**
+     * Applies the rotation to the given Data object.
+     *
+     * @param data Data object to encode.
+     * @return new Data object representing the result.
```
Comment (arraysius) [inline src/main/java/decodex/modules/rot/RotEncoder.java:31]: The description of the return data is not descriptive of the modification done. Consider `A Data object with its alphabetical characters rotated.`.

File: src/main/java/decodex/modules/rot/RotEncoder.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        return new Data(encodedString);
+    }
+
+    /**
+     * Checks if the given character is a lowercase alphabet.
+     *
+     * @param c The given character to check.
+     * @return True if the character is a lowercase alphabet.
+     */
+    private boolean isLowercaseAlphabet(char c) {
```
Comment (arraysius) [inline src/main/java/decodex/modules/rot/RotEncoder.java:58]: Consider using Java's `Character.isLowerCase()` instead.

File: src/main/java/decodex/modules/rot/RotEncoder.java [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        return LOWERCASE_A <= c && c <= LOWERCASE_Z;
+    }
+
+    /**
+     * Checks if the given character is an uppercase alphabet.
+     *
+     * @param c The given character to check.
+     * @return True if the character is an uppercase alphabet.
+     */
+    private boolean isUppercaseAlphabet(char c) {
```
Comment (arraysius) [inline src/main/java/decodex/modules/rot/RotEncoder.java:68]: Consider using Java's Character.isUpperCase() instead.

**Review by arraysius — CHANGES_REQUESTED** [review]
A change in the javadocs and make use of Java's built in functions instead of reinventing the wheel.

File: src/main/java/decodex/modules/rot/RotEncoder.java (hunk 1 above)
Comment (Kair0s3) [reply src/main/java/decodex/modules/rot/RotEncoder.java]: I think a suggestion would be `The Data object containing the rotated result`?

File: src/main/java/decodex/modules/rot/RotEncoder.java [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+    /**
+     * Rotates the given lowercase alphabet.
+     *
+     * @param c            The given lowercase alphabet.
+     * @param rotateOffset Number of positions to rotate the alphabet by.
+     * @return The updated alphabet.
+     */
+    private char rotateLowercaseAlphabet(char c, int rotateOffset) {
+        return (char) (Math.floorMod(c - LOWERCASE_A + rotateOffset, ALPHABETS_COUNT) + LOWERCASE_A);
```
Comment (Kair0s3) [inline src/main/java/decodex/modules/rot/RotEncoder.java:58]: Maybe this line could be broken down into smaller steps? (similarly with the other method)

File: src/test/java/decodex/modules/rot/RotEncoderTest.java [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    void run_nullBytes_noChange() {
+        Data testData = new Data("\00");
+        Data expectedData = new Data("\00");
+
+        RotEncoder rotEncoder = new RotEncoder(1);
+        Data resultData = rotEncoder.run(testData);
+
+        assertEquals(resultData.toString(), expectedData.toString());
+    }
+}
```
Comment (arraysius) [inline src/test/java/decodex/modules/rot/RotEncoderTest.java:86]: Missing new line at the end of file.

File: src/main/java/decodex/modules/rot/RotEncoder.java [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+     */
+    @Override
+    public Data run(Data data) {
+        StringBuilder encodedStringBuilder = new StringBuilder();
+
+        for (byte b : data.getRawBytes()) {
+            char updatedChar = (char) b;
+            if (Character.isLowerCase((char) b)) {
+                updatedChar = rotateLowercaseAlphabet((char) b, rotateOffset);
+
```
Comment (arraysius) [inline src/main/java/decodex/modules/rot/RotEncoder.java:39]: Empty line can be removed.

**Review by arraysius — CHANGES_REQUESTED** [review]
Minor changes.

**Review by arraysius — APPROVED** [review]
LGTM

**Review by Kair0s3 — APPROVED** [review]
LGTM

**Pull request merged**